mp/src/game/client/momentum/c_mom_player.h

```diff
@@ -6,6 +6,42 @@
 class C_MomentumOnlineGhostEntity;
 class C_MomentumReplayGhostEntity;
 
+struct SurfaceInteraction
+{
+    enum Type
+    {
+        TYPE_LEAVE = 0,
+        TYPE_FLOOR,
+        TYPE_WALL,
+        TYPE_CEILING,
+        TYPE_LAND,
+        TYPE_GROUNDED,
+        TYPE_COUNT
+    };
+
+    enum Action
+    {
+        ACTION_LEAVE = 0,
+        ACTION_KNOCKBACK,
+        ACTION_WALK,
+        ACTION_DUCKWALK,
+        ACTION_JUMP,
+        ACTION_DUCKJUMP,
+        ACTION_CTAP,
+        ACTION_COLLISION,
+        ACTION_SLIDE,
+        ACTION_EDGEBUG,
+        ACTION_LAND,
+        ACTION_GROUNDED,
+    };
+
+    int tick;
+    trace_t trace;
+    Vector velocity;
+    Action action;
+};
+typedef SurfaceInteraction SurfInt;
```

Comment:
This should be in the mom_run_entity header as it's shared on server/client and may be useful to have for ghosts down the line.